nsv13/code/modules/munitions/ship_weapons/energy_weapons/subspacecooling.dm

```diff
@@ -0,0 +1,107 @@
+/obj/machinery/cooling
+	name = "subspace unit"
+	desc = "A subspace unit."
+	icon = 'nsv13/icons/obj/subspace.dmi'
+	icon_state = "cooler"
+	circuit = /obj/item/circuitboard/machine
+	bound_width = 32
+	pixel_x = 0
+	pixel_y = 0
+	idle_power_usage =  2000
+	var/obj/machinery/ship_weapon/energy/parent
+	var/on = FALSE
+	density = TRUE
+	critical_machine = TRUE
+
+/obj/machinery/cooling/cooler
+	name = "subspace cooling unit"
+	desc = "A cooling unit that dumps the massive amounts of heat energy weapons generate into subspace."
+	circuit = /obj/item/circuitboard/machine/cooling
+
+
+
+/obj/machinery/cooling/cooler/Initialize(mapload)
+	. = ..()
+	var/obj/machinery/ship_weapon/energy/E = locate(/obj/machinery/ship_weapon/energy) in orange(1, src) //I have no idea what I'm doing and this causes errors so
+	E.coolers |= src
+	parent = E
```

Comment:
Unhandled references. Null parent on own `Destroy` and also remove from lists / clear parent relation if the parent object is being `Destroy()`ed.

Reply:
It does that in the below lines doesn't it?

Reply:
> Should I also put parent = null below parent.coolers = null?

`parent = null` should come after you do things related to the parent in the `Destroy`, yes,

However you should not null `parent.coolers`here, are parent still exists, that part is correct; What you are missing is handling for if parent is being deleted.